### PR TITLE
B-21492 Admin User can assign Secondary Transportation Office Assignments 

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Allows admin users to assign up to two transportation offices to office users.

- Adds TransportationOfficeAssignments slice to OfficeUser model
- Adds TransportationOfficeAssignments slice to UserIdentify model
- Adds TransportationOfficeAssignments array to admin create/update office user payloads
- Remove TransportationOfficeId from admin CREATE/UPDATE (but not GET) office user payloads
- Adds TransportationOfficeAssignments array to admin GET office user payloads
- ...

Hopefully this sets up the follow on task to be pretty easy.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the admin app.
2. Login as a new or existing admin user.
3. Create a new office user, expect the UI for assigning a transportation office to be slightly different.
4. Test out any combination of one or two transportation office assignments then modify by adding/remove/changing assignments a few times.
5. Ensure the designated primary office is displayed correctly on the list of office user and that the assignments you created display as you created them when you re-open the office user form.
6. Try to break it. The validation message refresh is a bit wonky but that seems to be a quirk of the react-admin library. That shouldn't affect your ability to submit valid data or not submit invalid data.

If you wish to verify the creation of your transportation office assignments in the DB the following query may be helpful.
`SELECT * FROM public.transportation_office_assignments toas
JOIN transportation_offices tos ON toas.transportation_office_id = tos.id
JOIN office_users ou ON ou.id = toas.id
WHERE ou.last_name = 'O''Testoghue'`

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-10-29 at 6 03 40 PM

### CODE REVIEW

**Review by antgmann — APPROVED** [review]
ANDI Screenshot, only thing that doesn't meet compliance is the already-present header with white text on light blue background. Code looks good. Haven't been able to break it yet
image

File: pkg/services/requested_office_users/requested_office_user_updater.go [hunk 1]
```diff
@@ -17,8 +17,22 @@ type requestedOfficeUserUpdater struct {
...
+	// filters := []services.QueryFilter{query.NewQueryFilter("id", "=", id.String())}
+	// err := o.builder.FetchOne(appCtx, &officeUser, filters)
+
+	requestedOfficeUserFetcher := NewRequestedOfficeUserFetcherPop()
+	officeUser, err := requestedOfficeUserFetcher.FetchRequestedOfficeUserByID(appCtx, id)
+
+	// err := appCtx.DB().Eager("TransportationOffice").Find(&officeUser, id)
+	// if err != nil {
+	// 	switch err {
+	// 	case sql.ErrNoRows:
```
[the remark this answers is not in the document]
Comment (ajlusk) [reply pkg/services/requested_office_users/requested_office_user_updater.go]: Resolved. Removed changes to this file and a few similar changes. They were remnants from before I figured out how to get the existing query builder to understand the relationship between the office user and transportation office assignment model.

**Review by CoryKleinjanCACI — APPROVED** [review]
Everything appears to work as expected and code changes look good.

**Review by r-mettler — APPROVED** [review]
ANDI is clean, tests passed, I was able to add a primary and a secondary office. I get errors if I try and add more than two offices. I like the UI you added for adding the offices. We discussed that the PrimaryOffice in the transportation_office_assignments needs to be made a *bool since its a nullable field in the DB. This was missed by me when I first created the model but will be added in the downstream to this series.

**Pull request merged**